### ISSUE TITLE
[Accessibility]: Search results syntax highlighting has insufficient contrast

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34739]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

Two main issues:

- Search results syntax highlighting themes result in insufficient text contrast in both light and dark mode, for both file results, diff results, and commit results
    - Examples of dark mode low contrast:
        - .theme-dark .hl-markup.hl-list 
        - .theme-dark .hl-comment, .theme-dark .hl-punctuation.hl-definition.hl-comment 
        - .theme-dark .hl-source.hl-diff .hl-meta.hl-range
    - Examples of light mode low contrast:
        - .theme-light .hl-source, .theme-light .hl-text 
        - .theme-light .hl-source.hl-diff .hl-meta.hl-range, .theme-light .hl-source.hl-perl .hl-support.hl-function 
        - .theme-light .hl-source.hl-diff .hl-entity.hl-name.hl-section
- Match highlight background colors results in insufficient text contrast in light and dark mode

### Expected behavior

Both the syntax highlighting themes and the match highlight are color contrast compliant, in both light and dark mode

### Additional details

Dark mode:

![image](https://user-images.githubusercontent.com/206864/166080790-ff59222b-b3d5-473b-bd5c-6804ccf366ce.png)

Light mode:

![image](https://user-images.githubusercontent.com/206864/166080808-da0fc55f-faed-49c5-aa60-0ae5b6bc34f7.png)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by umpox** [comment]
@limitedmage, I have started working on this with design, related issue we found here https://github.com/sourcegraph/sourcegraph/issues/36251

We have changes ready that make our syntax highlighting compliant

I'll assign myself to this, LMK if that's not OK!

**Comment by umpox** [comment]
The work to fix this has been merged in https://github.com/sourcegraph/sourcegraph/pull/37385. It is currently feature flagged so we can test internally before enabling it site-wide.

We have an issue (https://github.com/sourcegraph/sourcegraph/issues/37535) to enable it fully over the coming weeks, closing this in favour of that!
